test/extended/cmd.sh

```diff
@@ -71,75 +72,69 @@ popd
 
 
 DOCKER_CONFIG_JSON=${HOME}/.docker/config.json
-oc new-project dc-ns
-oc delete all --all
-oc delete secrets --all
-oc secrets new image-ns-pull .dockerconfigjson=${DOCKER_CONFIG_JSON}
-oc secrets new-dockercfg image-ns-pull-old --docker-email=fake@example.org --docker-username=imagensbuilder --docker-server=${DOCKER_REGISTRY} --docker-password=${IMAGE_NS_TOKEN}
-
-oc process -f test/extended/fixtures/image-pull-secrets/pod-with-no-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - 
-wait_for_command "oc describe pod/no-pull-pod | grep 'Back-off pulling image'" 30*TIME_SEC
-oc delete pods --all
-
-# TODO remove sleeps once jsonpath stops panicing.  The code still works without the sleep, it just looks nasty
-
-oc process -f test/extended/fixtures/image-pull-secrets/pod-with-new-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - 
-sleep 1
-wait_for_command "oc get pods/new-pull-pod -o jsonpath='{.status.containerStatuses[0].imageID}' | grep 'docker'" 30*TIME_SEC
-oc delete pods --all
-docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest
-
-
-oc process -f test/extended/fixtures/image-pull-secrets/pod-with-old-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - 
-sleep 1
-wait_for_command "oc get pods/old-pull-pod -o jsonpath={.status.containerStatuses[0].imageID} | grep 'docker'" 30*TIME_SEC
-oc delete pods --all
-docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest
-
-oc process -f test/extended/fixtures/image-pull-secrets/dc-with-old-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - 
-sleep 4
-wait_for_command "oc get pods/my-dc-old-1-prehook -o jsonpath={.status.containerStatuses[0].imageID} | grep 'docker'" 30*TIME_SEC
-oc delete all --all
-docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest
-
-oc process -f test/extended/fixtures/image-pull-secrets/dc-with-new-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - 
-sleep 4
-wait_for_command "oc get pods/my-dc-1-prehook -o jsonpath={.status.containerStatuses[0].imageID} | grep 'docker'" 30*TIME_SEC
-oc delete all --all
-docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest
+os::cmd::expect_success 'oc new-project dc-ns'
+os::cmd::expect_success 'oc delete all --all'
+os::cmd::expect_success 'oc delete secrets --all'
+os::cmd::expect_success "oc secrets new image-ns-pull .dockerconfigjson=${DOCKER_CONFIG_JSON}"
+os::cmd::expect_success "oc secrets new-dockercfg image-ns-pull-old --docker-email=fake@example.org --docker-username=imagensbuilder --docker-server=${DOCKER_REGISTRY} --docker-password=${IMAGE_NS_TOKEN}"
+
+os::cmd::expect_success "oc process -f test/extended/fixtures/image-pull-secrets/pod-with-no-pull-secret.yaml --value=DOCKER_REGISTRY=${DOCKER_REGISTRY} | oc create -f - "
+os::cmd::try_until_text 'oc describe pod/no-pull-pod' 'Back-off pulling image' 30*TIME_SEC
```

Comment:
Depending on how paranoid you are, you might want to quote `30*TIME_SEC` to guard against `nullglob`'s or `failglob`'s being set.